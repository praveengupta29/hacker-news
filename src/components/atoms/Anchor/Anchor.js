import React from 'react';
import PropTypes from 'prop-types';

import { buildUrl } from '../../../commons/utils/url';

const Anchor = ({
  children,
  to,
  className,
  handleLinkClick,
  target,
  ...others
}) => (
  <a
    {...others}
    href={buildUrl(to)}
    className={className}
    onClick={handleLinkClick}
    target={target}
  >
    {children}
  </a>
);

Anchor.defaultProps = {
  className: '',
  target: '_self',
  handleLinkClick: () => {},
  to: '#',
};

Anchor.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  handleLinkClick: PropTypes.func,
  to: PropTypes.any,
  target: PropTypes.oneOf(['', '_self', '_blank', '_parent', '_top']),
};

export default Anchor;
