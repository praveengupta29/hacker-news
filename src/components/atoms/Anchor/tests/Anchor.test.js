import React from 'react';
import { shallow } from 'enzyme';

import Anchor from '../Anchor';

describe('Anchor Component', () => {
  let anchor;

  test('should render correctly and match snapshot', () => {
    const props = {
      children: 'Sample',
      to: '/',
      onClick: () => {},
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with object destination', () => {
    const props = {
      children: 'Sample',
      to: {
        pathname: 'http://abc.com',
        query: {
          a: 1,
          b: 2,
        },
      },
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with object destination and no query param', () => {
    const props = {
      children: 'Sample',
      to: {
        pathname: 'http://abc.com',
      },
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
});
