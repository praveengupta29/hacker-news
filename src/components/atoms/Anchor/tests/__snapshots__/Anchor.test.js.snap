// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
<a
  className=""
  href="/"
  onClick={[Function]}
  target="_self"
>
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with object destination 1`] = `
<a
  className=""
  href="undefined?a=1&b=2"
  onClick={[Function]}
  target="_self"
>
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with object destination and no query param 1`] = `
<a
  className=""
  onClick={[Function]}
  target="_self"
>
  Sample
</a>
`;
